store/src/java-test/com/zimbra/cs/account/ldap/LdapProvisioningTest.java

```diff
@@ -0,0 +1,99 @@
+package com.zimbra.cs.account.ldap;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.MailboxTestUtil;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+/**
+ * Unit test for {@link LdapProvisioning}.
+ *
+ */
+public class LdapProvisioningTest {
+
+  @BeforeClass
+  public static void init() throws Exception {
+    MailboxTestUtil.initServer();
+    Provisioning prov = Provisioning.getInstance();
+    HashMap<String,Object> attrs;
+    attrs = new HashMap<>();
+    attrs.put(Provisioning.A_zimbraId, UUID.randomUUID().toString());
+    attrs.put(Provisioning.A_sn, "Leesa");
+    attrs.put(Provisioning.A_cn, "Natalie Leesa");
+    attrs.put(Provisioning.A_initials, "James");
+    attrs.put(Provisioning.A_mail, "natalie.leesa@example.com");
+    prov.createAccount("natalie.leesa@example.com", "testpassword", attrs);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    MailboxTestUtil.clearData();
+  }
+
+  /**
+   * This test is to validate functionality of
+   * ({@link com.zimbra.cs.account.ldap.LdapProvisioning#validatePasswordEntropyForPersonalData(String, Account)})
+   * @throws ServiceException */
+  @Test
+  public void shouldFindPersonalDataInPassword() throws ServiceException {
+
+    //Account acct = Provisioning.getInstance().get(AccountBy.id, "natalie.leesa@example.com");
+    Account acct = Provisioning.getInstance().getAccount("natalie.leesa@example.com");
+
+    //fake passwords for test against created user account
+    String[] testPasswords = {"natalie", "Natalie123", "Leesa34", "example01", "Leesa.Example", "eXamPle", "2022james"};
+
+    // get possible personal data for account
+    Optional<String> sName =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_sn)).orElse(""));
+    Optional<String> cName =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_cn)).orElse(""));
+    Optional<String> initials =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_initials)).orElse(""));
+    Optional<String> email =
+        Optional.of(Optional.ofNullable(acct.getAttr(Provisioning.A_mail)).orElse(""));
```

Comment:
Why this? You can simply do this:

```suggestion
    Optional<String> sName = Optional.ofNullable(acct.getAttr(Provisioning.A_sn));
    Optional<String> cName = Optional.ofNullable(acct.getAttr(Provisioning.A_cn));
    Optional<String> initials = Optional.ofNullable(acct.getAttr(Provisioning.A_initials));
    Optional<String> email = Optional.ofNullable(acct.getAttr(Provisioning.A_mail));
```

KISS

Reply:
I don't want to deal with noSuchElementException in my code below these initializations. 
Obviously there could be 1 thousand other ways to keep it simple, but this one works for me. :)